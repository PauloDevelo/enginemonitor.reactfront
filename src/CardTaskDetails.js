import React from 'react';
import { 
    Button, 
	Card,
	CardBody,
	CardTitle,
	CardSubtitle,
	CardFooter,
	CardText,
	Badge
} from 'reactstrap';

import { FormattedMessage } from 'react-intl';

import PropTypes from 'prop-types';

import { getContext, getScheduleText } from './TaskHelper'; 

import edittaskmsg from "./ModalEditTask.messages";

import './CardTaskDetails.css';

function getBadgeText(level){
	if(level === 1){
		return 'Done'
	}
	else if(level === 2){
		return 'Soon'
	}
	else{
		return 'Todo'
	}
}

const CardTaskDetails = ({task, next, prev, toggleModal, nextVisibility, prevVisibility, toggleAckModal, classNames}) => {
    if (task === undefined){
        return <Card className="p-2 m-2 border border-primary rounded shadow"/>;
    }

    const cursorPointerStyle = {
        cursor: 'pointer',
    };

    var badgeText = getBadgeText(task.level);
    var badgeContext = getContext(task.level);

    var title = getScheduleText(task);

    var descriptionFormatted = task.description.replace(/\n/g,"<br />");

    var prevClassNames = "card-control-prev-icon";
    if(!prevVisibility)
        prevClassNames += ' invisible';

    var nextClassNames = "card-control-next-icon";
    if(!nextVisibility)
        nextClassNames += ' invisible';

    return(<Card className={classNames}>
            <CardBody className="d-flex p-0">
                    <div className="p-2" onClick={prev} style={cursorPointerStyle}><div className={prevClassNames}></div></div>
                    <div className="p-2 flex-grow-1">
                        <CardTitle>{task.name} <Badge color={badgeContext} pill>{badgeText}</Badge></CardTitle>
                        <CardSubtitle>{title}</CardSubtitle>
                        <CardText dangerouslySetInnerHTML={{ __html: descriptionFormatted }}></CardText>
                    </div>
                    <div className="p-2" onClick={next} style={cursorPointerStyle}><div className={nextClassNames}></div></div>
            </CardBody>
            <CardFooter className='pl-5 pr-5'>
                <Button color='primary' className='float-left' onClick={toggleModal}><FormattedMessage {...edittaskmsg.edit} /></Button>
                <Button color='primary' className='float-right' onClick={toggleAckModal}><FormattedMessage {...edittaskmsg.ack} /></Button>
            </CardFooter>
        </Card>
    );
}

CardTaskDetails.propTypes = {
    toggleModal: PropTypes.func.isRequired,
    next: PropTypes.func.isRequired,
    nextVisibility: PropTypes.bool.isRequired,
    prev: PropTypes.func.isRequired,
    prevVisibility: PropTypes.bool.isRequired,
    task: PropTypes.object,
    toggleAckModal: PropTypes.func.isRequired,
    classNames: PropTypes.string
};

export default CardTaskDetails;