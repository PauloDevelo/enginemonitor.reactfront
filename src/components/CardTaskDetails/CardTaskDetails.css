
.card-control-prev-icon{
	background-image:url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23000' viewBox='0 0 8 8'%3E%3Cpath d='M5.25 0l-4 4 4 4 1.5-1.5-2.5-2.5 2.5-2.5-1.5-1.5z'/%3E%3C/svg%3E");
    width: 20px;
    height: 20px;
    opacity: 0.2;
}

.card-control-prev-icon:hover {
    opacity: 1.0;
}

.card-control-next-icon{
    background-image:url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23000' viewBox='0 0 8 8'%3E%3Cpath d='M2.75 0l-1.5 1.5 2.5 2.5-2.5 2.5 1.5 1.5 4-4-4-4z'/%3E%3C/svg%3E");
    width: 20px;
    height: 20px;
    opacity: 0.2;
}

.card-control-next-icon:hover {
    opacity: 1.0;
}

.clickable {
    cursor: pointer;
}

.hover { 
    z-index: -1; 
    opacity: 0.5; 
    transform-style:all; 
    transition-duration: 1s; 
} 