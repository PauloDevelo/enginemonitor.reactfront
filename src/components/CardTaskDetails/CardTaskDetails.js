import React from 'react';
import { Button, Card, CardBody, CardTitle, CardSubtitle, CardFooter, CardText, Badge } from 'reactstrap';
import { faEdit } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import PropTypes from 'prop-types';

import { getContext, getScheduleText } from '../../helpers/TaskHelper'; 

import './CardTaskDetails.css';
import '../../style/transition.css';

function getBadgeText(level){
	if(level === 1){
		return 'Done'
	}
	else if(level === 2){
		return 'Soon'
	}
	else{
		return 'Todo'
	}
}

const CardTaskDetails = ({task, next, prev, toggleModal, nextVisibility, prevVisibility, classNames}) => {
    if (task === undefined){
        return <Card className={classNames}/>;
    }

    const cursorPointerStyle = {
        cursor: 'pointer',
    };

    var badgeText = getBadgeText(task.level);
    var badgeContext = getContext(task.level);

    var title = getScheduleText(task);

    var descriptionFormatted = task.description.replace(/\n/g,"<br />");

    var prevClassNames = "card-control-prev-icon";
    if(!prevVisibility)
        prevClassNames += ' invisible';

    var nextClassNames = "card-control-next-icon";
    if(!nextVisibility)
        nextClassNames += ' invisible';

    return(
        <Card className={classNames}>        
            <CardBody className="d-flex p-0">
                    <div className="p-2" onClick={prev} style={cursorPointerStyle}><div className={prevClassNames}></div></div>
                        <TransitionGroup className="p-2 flex-grow-1">
                            <CSSTransition key={task._id} timeout={250} classNames="card" >
                                <div >
                                    <CardTitle>{task.name} <Badge color={badgeContext} pill>{badgeText}</Badge></CardTitle>
                                    <CardSubtitle>{title}</CardSubtitle>
                                    <CardText dangerouslySetInnerHTML={{ __html: descriptionFormatted }}></CardText>
                                </div>
                            </CSSTransition>
                        </TransitionGroup>
                    <div className="p-2" onClick={next} style={cursorPointerStyle}><div className={nextClassNames}></div></div>
            </CardBody>
            <CardFooter className='pl-5 pr-5'>
                <Button color='light' className='float-left' onClick={toggleModal}><FontAwesomeIcon icon={faEdit} /></Button>
            </CardFooter>
        </Card>
    );
}

CardTaskDetails.propTypes = {
    task: PropTypes.object,
    next: PropTypes.func.isRequired,
    prev: PropTypes.func.isRequired,
    toggleModal: PropTypes.func.isRequired,
    nextVisibility: PropTypes.bool.isRequired,
    prevVisibility: PropTypes.bool.isRequired,
    classNames: PropTypes.string
};

export default CardTaskDetails;