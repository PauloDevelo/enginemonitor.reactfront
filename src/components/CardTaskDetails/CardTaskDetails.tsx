import React, {Fragment, useEffect} from 'react';
import { Button, Card, CardBody, CardTitle, CardSubtitle, CardFooter, CardText, Badge } from 'reactstrap';
import { faEdit } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import PropTypes from 'prop-types';

import { useEditModal } from '../../hooks/EditModalHook';

import { getContext, getScheduleText, getBadgeText } from '../../helpers/TaskHelper'; 

import ModalEditTask from '../ModalEditTask/ModalEditTask';

import './CardTaskDetails.css';
import '../../style/transition.css';
import { EquipmentModel, TaskModel } from '../../types/Types';

type Props = {
    equipment?: EquipmentModel, 
    tasks: TaskModel[], 
    currentTask?: TaskModel, 
    onTaskChanged: React.MutableRefObject<(task: TaskModel) => void>, 
    onTaskDeleted: (task: TaskModel) => void, 
    changeCurrentTask: (task: TaskModel | undefined) => void, 
    classNames?: string
}

const CardTaskDetails = ({equipment, tasks, currentTask, onTaskChanged, onTaskDeleted, changeCurrentTask, classNames}: Props) => {
    const modalHook = useEditModal(currentTask);

    const getTaskIndex = (task: TaskModel | undefined):number => {
        return task === undefined ? -1 : tasks.findIndex(t => t._uiId === task._uiId);
    }

    const taskIndex = getTaskIndex(currentTask);
    

    if (equipment === undefined || currentTask === undefined){
        return <Card className={classNames}/>;
    }

    const isPrevButtonVisible = ():boolean => (taskIndex > 0);
    const isNextButtonVisible = ():boolean => (taskIndex < tasks.length - 1);

    const nextTask = ():void => {
        if(isNextButtonVisible())
            changeCurrentTask(tasks[taskIndex + 1]);
    };
	const previousTask = ():void => {
        if(isPrevButtonVisible())
            changeCurrentTask(tasks[taskIndex - 1]);
    }

    const cursorPointerStyle = { cursor: 'pointer' };
    const badgeText = getBadgeText(currentTask.level);
    const badgeContext = getContext(currentTask.level);
    const title = getScheduleText(equipment, currentTask);
    const descriptionFormatted = currentTask.description.replace(/\n/g,"<br />");

    let prevClassNames = "card-control-prev-icon";
    if(!isPrevButtonVisible())
        prevClassNames += ' invisible';

    let nextClassNames = "card-control-next-icon";
    if(!isNextButtonVisible())
        nextClassNames += ' invisible';

    return(
        <Fragment>
            <Card className={classNames}>        
                <CardBody className="d-flex p-0">
                        <div className="p-2 button-previous-task" onClick={previousTask} style={cursorPointerStyle}><div className={prevClassNames}></div></div>
                        <TransitionGroup className="p-2 flex-grow-1">
                            <CSSTransition key={currentTask._uiId} timeout={250} classNames="card" >
                                <div >
                                    <CardTitle>{currentTask.name} <Badge color={badgeContext} pill>{badgeText}</Badge></CardTitle>
                                    <CardSubtitle>{title}</CardSubtitle>
                                    <CardText dangerouslySetInnerHTML={{ __html: descriptionFormatted }}></CardText>
                                </div>
                            </CSSTransition>
                        </TransitionGroup>
                        <div className="p-2 button-next-task" onClick={nextTask} style={cursorPointerStyle}><div className={nextClassNames}></div></div>
                </CardBody>
                <CardFooter className='pl-5 pr-5'>
                    <Button color='light' className='float-left' onClick={modalHook.toggleModal} aria-label="Edit"><FontAwesomeIcon icon={faEdit} /></Button>
                </CardFooter>
            </Card>

            {modalHook.editModalVisibility && <ModalEditTask  equipment={equipment}
                            task={currentTask}
                            onTaskSaved={onTaskChanged.current} 
                            onTaskDeleted={onTaskDeleted}
                            visible={modalHook.editModalVisibility} 
                            toggle={modalHook.toggleModal} 
                            className='modal-dialog-centered'/>}
        </Fragment>
    );
}

CardTaskDetails.propTypes = {
    equipment: PropTypes.object,
    tasks: PropTypes.array,
    currentTask: PropTypes.object,
    onTaskChanged: PropTypes.object.isRequired,
    onTaskDeleted: PropTypes.func.isRequired,
    changeCurrentTask: PropTypes.func.isRequired,
    classNames: PropTypes.string
};

export default React.memo(CardTaskDetails);