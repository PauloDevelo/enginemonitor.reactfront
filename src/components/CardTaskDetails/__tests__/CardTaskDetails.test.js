import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import CardTaskDetails from '../CardTaskDetails';

const mockConsoleMethod = (realConsoleMethod) => {
    const ignoredMessages = [
        "[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry. Using default message as fallback."
    ];

    return (message, ...args) => {
        const containsIgnoredMessage = ignoredMessages.some((ignoredMessage) => message.includes(ignoredMessage));

        if (!containsIgnoredMessage) {
        realConsoleMethod(message, ...args);
        }
    };
};
  
// Suppress console errors and warnings to avoid polluting output in tests.
console.warn = jest.fn(mockConsoleMethod(console.warn));
console.error = jest.fn(mockConsoleMethod(console.error));

describe("CardTaskDetails", () => {
    const equipment = {
        _uiId: '1234',
        name: 'engine',
        brand: 'nanni',
        model: 'N3.30',
        age: 2563,
        installation: new Date('2011-02-22T16:00:00.000Z')
    }

    const task1 = {
        _uiId: 'task1',
        name: 'taskname1',
        periodInMonth: 12,
        description: 'task1 description',
        nextDueDate: new Date('2011-02-22T16:00:00.000Z'),
        usagePeriodInHour: 200,
        usageInHourLeft: 20,
        level: 1
    }

    const task2 = {
        _uiId: 'task2',
        name: 'taskname2',
        periodInMonth: 24,
        description: 'task2 description',
        nextDueDate: new Date('2018-02-22T16:00:00.000Z'),
        usagePeriodInHour: 400,
        usageInHourLeft: 20,
        level: 3
    }

    const tasks = [task1, task2];
    const onTaskChangedMock = jest.fn();
    const onTaskDeletedMock = jest.fn();
    const changeCurrentTaskMock = jest.fn();

    beforeEach(() => {
        onTaskChangedMock.mockClear();
        onTaskDeletedMock.mockClear();
        changeCurrentTaskMock.mockClear();
    });

    it("Should render correctly even if the equipment is undefined", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={undefined} 
            tasks={[]} 
            currentTask={undefined}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Assert
        expect(wrapper).toMatchSnapshot();
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
    });

    it("Should render correctly even if the task array is empty", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={[]} 
            currentTask={undefined}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Assert
        expect(wrapper).toMatchSnapshot();
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
    });

    it("Should render correctly even if the current task is undefined", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={undefined}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Assert
        expect(wrapper).toMatchSnapshot();
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
    });

    it("Should render the task 1 details", () => {
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task1}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        expect(wrapper.find('.card-title').text()).toEqual(task1.name + ' Done');
        expect(wrapper.find('.badge-success').text()).toEqual("Done");
        expect(wrapper.find('.card-control-prev-icon').hasClass('invisible')).toEqual(true);
        expect(wrapper.find('.card-text').text()).toEqual(task1.description);
        expect(wrapper.find('.card-control-next-icon').hasClass('invisible')).toEqual(false);

        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
    });

    it("should stay the same with task1", () => {
        const tree = renderer.create(<CardTaskDetails 
                                        equipment={equipment} 
                                        tasks={tasks} 
                                        currentTask={task1}
                                        onTaskChanged={onTaskChangedMock}
                                        onTaskDeleted={onTaskDeletedMock}
                                        changeCurrentTask={changeCurrentTaskMock} 
                                    />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should render the task 2 details", () => {
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task2}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        expect(wrapper.find('.card-title').text()).toEqual(task2.name + ' ToDo');
        expect(wrapper.find('.badge-danger').text()).toEqual("ToDo");
        expect(wrapper.find('.card-control-prev-icon').hasClass('invisible')).toEqual(false);
        expect(wrapper.find('.card-text').text()).toEqual(task2.description);
        expect(wrapper.find('.card-control-next-icon').hasClass('invisible')).toEqual(true);

        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
    });

    it("should stay the same with task2", () => {
        const tree = renderer.create(<CardTaskDetails 
                                        equipment={equipment} 
                                        tasks={tasks} 
                                        currentTask={task2}
                                        onTaskChanged={onTaskChangedMock}
                                        onTaskDeleted={onTaskDeletedMock}
                                        changeCurrentTask={changeCurrentTaskMock} 
                                    />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should call changeCurrentTask after clicking on the next button", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task1}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Act
        wrapper.find('.button-next-task').simulate('click');

        // Assert
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(1);
        expect(changeCurrentTaskMock.mock.calls[0][0]).toBe(task2);
        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
    });

    it("Should not call changeCurrentTask after clicking on the next button because task2 is the last task", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task2}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Act
        wrapper.find('.button-next-task').simulate('click');

        // Assert
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
    });

    it("Should not call changeCurrentTask after clicking on the prev button because task1 is the first task", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task1}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Act
        wrapper.find('.button-previous-task').simulate('click');

        // Assert
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(0);
        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
    });

    it("Should call changeCurrentTask after clicking on the prev button", () => {
        // Arrange
        const wrapper = mount(<CardTaskDetails 
            equipment={equipment} 
            tasks={tasks} 
            currentTask={task2}
            onTaskChanged={onTaskChangedMock}
            onTaskDeleted={onTaskDeletedMock}
            changeCurrentTask={changeCurrentTaskMock} 
        />);

        // Act
        wrapper.find('.button-previous-task').simulate('click');

        // Assert
        expect(changeCurrentTaskMock).toHaveBeenCalledTimes(1);
        expect(changeCurrentTaskMock.mock.calls[0][0]).toBe(task1);
        expect(onTaskChangedMock).toHaveBeenCalledTimes(0);
        expect(onTaskDeletedMock).toHaveBeenCalledTimes(0);
    });
});