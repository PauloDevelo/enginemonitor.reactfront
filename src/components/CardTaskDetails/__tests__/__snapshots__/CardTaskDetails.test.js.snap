// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTaskDetails Should render correctly even if the current task is undefined 1`] = `
<CardTaskDetails
  changeCurrentTask={[MockFunction]}
  equipment={
    Object {
      "_id": "1234",
      "age": 2563,
      "brand": "nanni",
      "installation": 2011-02-22T16:00:00.000Z,
      "model": "N3.30",
      "name": "engine",
    }
  }
  onTaskChanged={[MockFunction]}
  onTaskDeleted={[MockFunction]}
  tasks={
    Array [
      Object {
        "_id": "task1",
        "description": "task1 description",
        "level": 1,
        "name": "taskname1",
        "nextDueDate": 2011-02-22T16:00:00.000Z,
        "periodInMonth": 12,
        "usageInHourLeft": 20,
        "usagePeriodInHour": 200,
      },
      Object {
        "_id": "task2",
        "description": "task2 description",
        "level": 3,
        "name": "taskname2",
        "nextDueDate": 2018-02-22T16:00:00.000Z,
        "periodInMonth": 24,
        "usageInHourLeft": 20,
        "usagePeriodInHour": 400,
      },
    ]
  }
>
  <div
    className="card"
  />
</CardTaskDetails>
`;

exports[`CardTaskDetails Should render correctly even if the equipment is undefined 1`] = `
<CardTaskDetails
  changeCurrentTask={[MockFunction]}
  onTaskChanged={[MockFunction]}
  onTaskDeleted={[MockFunction]}
  tasks={Array []}
>
  <div
    className="card"
  />
</CardTaskDetails>
`;

exports[`CardTaskDetails Should render correctly even if the task array is empty 1`] = `
<CardTaskDetails
  changeCurrentTask={[MockFunction]}
  equipment={
    Object {
      "_id": "1234",
      "age": 2563,
      "brand": "nanni",
      "installation": 2011-02-22T16:00:00.000Z,
      "model": "N3.30",
      "name": "engine",
    }
  }
  onTaskChanged={[MockFunction]}
  onTaskDeleted={[MockFunction]}
  tasks={Array []}
>
  <div
    className="card"
  />
</CardTaskDetails>
`;

exports[`CardTaskDetails should stay the same with task1 1`] = `
<div
  className="card"
>
  <div
    className="d-flex p-0 card-body"
  >
    <div
      className="p-2 button-previous-task"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <div
        className="card-control-prev-icon invisible"
      />
    </div>
    <div
      className="p-2 flex-grow-1"
    >
      <div>
        <div
          className="card-title"
        >
          taskname1
           
          <span
            className="badge badge-success badge-pill"
          >
            Done
          </span>
        </div>
        <div
          className="card-subtitle"
        >
          <span>
            <span>
              To do every 
            </span>
            <b>
              200
               h
               
            </b>
            <span>
               or every 
            </span>
            <b>
              <span>
                12 months
              </span>
            </b>
          </span>
        </div>
        <p
          className="card-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "task1 description",
            }
          }
        />
      </div>
    </div>
    <div
      className="p-2 button-next-task"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <div
        className="card-control-next-icon"
      />
    </div>
  </div>
  <div
    className="pl-5 pr-5 card-footer"
  >
    <button
      aria-label={null}
      className="float-left btn btn-light"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-edit fa-w-18 "
        data-icon="edit"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`CardTaskDetails should stay the same with task2 1`] = `
<div
  className="card"
>
  <div
    className="d-flex p-0 card-body"
  >
    <div
      className="p-2 button-previous-task"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <div
        className="card-control-prev-icon"
      />
    </div>
    <div
      className="p-2 flex-grow-1"
    >
      <div>
        <div
          className="card-title"
        >
          taskname2
           
          <span
            className="badge badge-danger badge-pill"
          >
            ToDo
          </span>
        </div>
        <div
          className="card-subtitle"
        >
          <span>
            <span>
              To do every 
            </span>
            <b>
              400
               h
               
            </b>
            <span>
               or every 
            </span>
            <b>
              <span>
                24 months
              </span>
            </b>
          </span>
        </div>
        <p
          className="card-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "task2 description",
            }
          }
        />
      </div>
    </div>
    <div
      className="p-2 button-next-task"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <div
        className="card-control-next-icon invisible"
      />
    </div>
  </div>
  <div
    className="pl-5 pr-5 card-footer"
  >
    <button
      aria-label={null}
      className="float-left btn btn-light"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-edit fa-w-18 "
        data-icon="edit"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;
